lib/network_utils.ml

```diff
@@ -484,7 +484,21 @@ module Ip = struct
 
   let get_mtu dev = int_of_string (List.hd (link dev "mtu"))
 
-  let get_mac dev = List.hd (link dev "link/ether")
+  let get_mac dev =
+    match link dev "link/ether" with
```

Comment:
This nested match looks awkward, but it's obvious that `link dev` is an expensive operation and should be called as little possible. So this looks good to me

Using a high-level function this could be done:
```ocaml
let mac =
  List.find_map_opt (fun name -> List.nth (link dev name) 0) ["link/ether"; "link/infiniband"]
in
let match mac with None -> raise err | Some mac -> mac
```

But we don't have List.find_map_opt available and don't think it's worth implementing it here :(

Reply:
I'm also not happy about this but had the same thought: I don't want to run the `ip` command when it it not needed. Otherwise I could have done `match link dev "link/ether", link dev "link/infiniband" with` and at least avoid the nesting. I can still do this if anybody would prefer it. I also like to keep backporting in mind and avoid code that makes that difficult.